VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -367,4 +375,73 @@ public void orientationChangeTest() {
                 activity.getResources().getConfiguration().orientation);
         assertOrientationSetup(activity);
     }
+
+    /**
+     * Testing creation of various fragments in the {@link CourseUnitNavigationActivity}'s
+     * ViewPager, by supplying its {@link CourseUnitPagerAdapter} with all possible
+     * {@link CourseComponent} models.
+     */
+    @Test
+    public void testUnitFragmentCreation() {
```

Comment:
Unfortunately, the `RobolectricGradleTestRunner` [doesn't support][1] parameterized test suites at the moment. However, the same effect could be achieved more or less by creating an assertion method that takes a parameter and reports it's values on failure, and then calling it iteratively with all the parameters from the actual test methods, which would also obviate the need to create a custom class for just one test method. I've implemented this general pattern a few times in the `BaseFragmentActivityTest` and elsewhere.

[1]: https://github.com/robolectric/robolectric/issues/1934